diff -uraN mate-sensors-applet-1.26.0/lib/sensors-applet-plugin.c omv-mate-sensors-applet-1.26.0/lib/sensors-applet-plugin.c
--- mate-sensors-applet-1.26.0/lib/sensors-applet-plugin.c	2021-08-05 10:07:02.000000000 +0200
+++ omv-mate-sensors-applet-1.26.0/lib/sensors-applet-plugin.c	2021-08-31 13:46:25.473993241 +0200
@@ -27,8 +27,6 @@
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
 
-extern const gchar *plugin_name;
-
 /* recursive function to find sensors in a given path */
 void sensors_applet_plugin_find_sensors(GList **sensors,
                                         const gchar *path,
@@ -63,7 +61,7 @@
 }
 
 /* for error handling */
-GQuark sensors_applet_plugin_error_quark(void) {
+GQuark sensors_applet_plugin_error_quark(const gchar *plugin_name) {
     static GQuark quark = 0;
 
     if (quark == 0) {
