Index: mate-sensors-applet-1.20.0/lib/sensors-applet-plugin.c
===================================================================
--- mate-sensors-applet-1.20.0/lib/sensors-applet-plugin.c
+++ mate-sensors-applet-1.20.0/lib/sensors-applet-plugin.c	2018-02-15 00:38:09.010446334 +0300
@@ -26,8 +26,6 @@
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
 
-extern const gchar *plugin_name;
-
 /* recursive function to find sensors in a given path */
 void sensors_applet_plugin_find_sensors(GList **sensors,
                                         const gchar *path,
@@ -61,7 +59,7 @@
 }
 
 /* for error handling */
-GQuark sensors_applet_plugin_error_quark(void) {
+GQuark sensors_applet_plugin_error_quark(const gchar *plugin_name) {
     static GQuark quark = 0;
 
     if (quark == 0) {
Index: mate-sensors-applet-1.20.0/sensors-applet/sensors-applet-plugin.h
===================================================================
--- mate-sensors-applet-1.20.0/sensors-applet/sensors-applet-plugin.h
+++ mate-sensors-applet-1.20.0/sensors-applet/sensors-applet-plugin.h	2018-02-15 00:41:31.166105683 +0300
@@ -21,10 +21,10 @@
 
 #include <sensors-applet/sensors-applet-sensor.h>
 
-GQuark sensors_applet_plugin_error_quark(void);
+GQuark sensors_applet_plugin_error_quark(const gchar *plugin_name);
 
 /* for error handling */
-#define SENSORS_APPLET_PLUGIN_ERROR (sensors_applet_plugin_error_quark())
+#define SENSORS_APPLET_PLUGIN_ERROR (sensors_applet_plugin_error_quark(plugin_name))
 
 
 /**
